src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import './TodoForm.scss';
+
+import { usersForSelect } from '../services/usersForSelect';
+
+import { Todo } from '../types/todo';
+import { getUserById } from '../services/userById';
+
+type Props = {
+  onAdd: (todo: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onAdd }) => {
+  // #region state
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+  // #endregion
+  // #region change handlers
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value.replace(/[^a-zA-Za-яА-Я0-9\s]/g, '');
+
+    setTitle(newTitle);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserError(false);
+  };
+  // #endregion
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title.trim());
+    setHasUserError(!userId);
+
+    if (!title.trim() || userId === 0) {
+      return;
+    }
+
+    onAdd({
+      id: 0,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    });
+
+    reset();
+  };
+
+  return (
+    <form
+      action="/api/todos"
+      method="POST"
+      onSubmit={handleSubmit}
+    >
+      <div className="field">
+        <label
+          className="label"
+          htmlFor="todo-title"
+        >
+          {'Title: '}
+        </label>
+
+        <input
+          id="todo-title"
+          type="text"
+          data-cy="titleInput"
+          placeholder="Enter a title"
+          value={title}
+          onChange={handleTitleChange}
+        />
+        {hasTitleError && (
+          <span className="error">
+            Please enter a title
+          </span>
+        )}
+      </div>
+
+      <div className="field">
+        <label
+          className="label"
+          htmlFor="todo-user"
+        >
+          {'User: '}
```

Comment:
it's bad practice to adding spaces this way, you can use margins and paddings instead